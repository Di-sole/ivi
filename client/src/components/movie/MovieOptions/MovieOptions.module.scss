.container {
    margin: 32px 0px 8px;
    border-top: 1px solid #ffffff29;
    border-bottom: 1px solid #ffffff29;
    padding: 16px 0px 32px;
    font-size: 15px;
    color: #a5a1b2;

    &__block {
        display: flex;
        align-items: center;
        padding: 12px 0;
        border-bottom: 1px solid #ffffff29;

        &:last-child {
            border: none;
        }
    }

    &__title {

        &_mobile {
            display: none;
        }
    }

    &__quality {
        color: #ffffff66;
    }

    &__value {
        margin-top: 4px;
        color: #fff;
    }

    &__badgesContainer {
        display: flex;
    }

    &__badge {
        margin-left: 8px;
    }
}

@media (max-width: 1159px) {
    .container {
        margin: 0;
        border: 0;
        padding: 0px 0px 24px;
        width: 100%;

        &__block {
            justify-content: space-between;
        }

        &__title {

            &_full {
                display: none;
            }

            &_mobile {
                display: flex;
            }
        }

        &__value {
            margin-top: 0px;
        }

        &__badge {
            margin-top: 0px;
        }
    }
}


@media (max-width: 599px) {
    .container {
        padding: 0px 0px 16px;
    }
}
