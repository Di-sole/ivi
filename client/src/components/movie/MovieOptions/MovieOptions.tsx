import styles from './MovieOptions.module.scss';
import { IMovie } from '../../squareCard/SquareCardsList/Temp/IMovie';
import TextBadge from '@/components/UI/badges/TextBadge/TextBadge';

interface MovieOptionsProps {
    movie: IMovie;
}

const MovieOptions = ({ movie }: MovieOptionsProps) => {
    return (
        <div className={styles.options}>
            <div className={styles.options__block}>
                <div className={styles.options__blockTitle}>Языки</div>
                <div className={styles.options__blockValue}>Русский, Английский</div>
            </div>
            <div className={styles.options__block}>
                <div className={styles.options__blockTitle}>Субтитры</div>
                <div className={styles.options__blockValue}>Русский</div>
            </div>
            <div className={styles.options__block}>
                <div className={[styles.options__blockTitle, styles.options__blockTitle_full].join(' ')}>
                    Изображение и звук.
                    <span>Фактическое качество зависит от устройства и ограничений правообладателя.</span>
                </div>
                <div className={[styles.options__blockTitle, styles.options__blockTitle_mobile].join(' ')}>
                    Качество
                </div>
                <div className={styles.options__badge}>
                    {movie.displays.map((display) => (
                        <div className={styles.options__badgeItem}>
                            <TextBadge key={display} text={display} />
                        </div>
                    ))}
                </div>
            </div>
        </div>
    );
};

export default MovieOptions;
