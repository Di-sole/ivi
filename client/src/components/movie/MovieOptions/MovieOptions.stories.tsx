import { Meta, StoryObj } from '@storybook/react';
import MovieOptions from './MovieOptions';
import '../../../styles/nullstyle.scss';
import '../../../styles/globals.scss';
import './MovieOptions.module.scss';
import { IMovie } from '../../movie/movieMedallion/MovieMedallionsList/Temp/IMovie';

import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';

const meta: Meta<typeof MovieOptions> = {
    title: 'Information/Options',
    component: MovieOptions,
    parameters: {
        viewport: {
            viewports: INITIAL_VIEWPORTS,
            default: 'desktop',
        },
    },
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof MovieOptions>;

const data: IMovie = {
    id: 1,
    title: 'Название',
    year: 1900,
    ageLimit: '0+',
    time: 'Длительность',
    production: 'Производство',
    genres: ['Жанр', 'Жанр', 'Жанр'],
    displays: ['FullHD', 'HD', '1080', '720'],
    voiceActing: ['Рус', 'Eng'],
    subtitles: ['Рус'],
    subtitlesFull: ['Русский'],
    description: [],
    language: ['Русский', 'Английский'],
    raiting: 9.5,
    nameraiting: 'Рейтинг',
    actors: [],
};

export const FullScreen: Story = {
    args: {
        movie: data,
    },
};

export const MobileScreen: Story = {
    parameters: {
        viewport: {
            defaultViewport: 'iphonex',
        },
    },
    args: {
        movie: data,
    },
};
