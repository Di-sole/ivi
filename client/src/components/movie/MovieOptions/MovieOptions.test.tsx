import { render, screen } from '@testing-library/react';
import MovieOptions from './MovieOptions';
import { movies } from '../movieMedallion/MovieMedallionsList/Temp/Movie.data';

describe('MOVIEOPTIONS SNAPSHOTS TESTS', () => {
    test('MovieOptions should not be changed', () => {
        render(<MovieOptions movie={movies[0]} />);

        const div = screen.getByRole('div-options');
        expect(div).toMatchSnapshot();
    });
});
