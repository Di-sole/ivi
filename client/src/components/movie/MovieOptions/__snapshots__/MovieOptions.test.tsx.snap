// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEOPTIONS SNAPSHOTS TESTS MovieOptions should not be changed 1`] = `
<div
  class="container"
  role="div-options"
>
  <div
    class="container__block"
  >
    <div
      class="container__title"
    >
      Языки
    </div>
    <div
      class="container__value"
    >
      Русский, Английский
    </div>
  </div>
  <div
    class="container__block"
  >
    <div
      class="container__title"
    >
      Субтитры
    </div>
    <div
      class="container__value"
    >
      Русский
    </div>
  </div>
  <div
    class="container__block"
  >
    <div
      class="container__title container__title_full"
    >
      Изображение и звук.
      <span
        class="container__quality"
      >
        Фактическое качество зависит от устройства и ограничений правообладателя.
      </span>
    </div>
    <div
      class="container__title container__title_mobile"
    >
      Качество
    </div>
    <div
      class="container__badgesContainer"
    >
      <div
        class="container__badge"
      >
        <div
          class="textBadge"
        >
          FullHD
        </div>
      </div>
      <div
        class="container__badge"
      >
        <div
          class="textBadge"
        >
          HD
        </div>
      </div>
      <div
        class="container__badge"
      >
        <div
          class="textBadge"
        >
          1080
        </div>
      </div>
      <div
        class="container__badge"
      >
        <div
          class="textBadge"
        >
          720
        </div>
      </div>
    </div>
  </div>
</div>
`;
